src/cascadia/TerminalSettingsEditor/Profiles.h

```diff
@@ -5,143 +5,25 @@
-        OBSERVABLE_PROJECTED_SETTING(_profile, ScrollState);
-        OBSERVABLE_PROJECTED_SETTING(_profile, Padding);
-        OBSERVABLE_PROJECTED_SETTING(_profile, Commandline);
-        OBSERVABLE_PROJECTED_SETTING(_profile, StartingDirectory);
-        OBSERVABLE_PROJECTED_SETTING(_profile, AntialiasingMode);
-        OBSERVABLE_PROJECTED_SETTING(_profile.DefaultAppearance(), Foreground);
-        OBSERVABLE_PROJECTED_SETTING(_profile.DefaultAppearance(), Background);
-        OBSERVABLE_PROJECTED_SETTING(_profile.DefaultAppearance(), SelectionBackground);
-        OBSERVABLE_PROJECTED_SETTING(_profile.DefaultAppearance(), CursorColor);
-        OBSERVABLE_PROJECTED_SETTING(_profile.DefaultAppearance(), Opacity);
-        OBSERVABLE_PROJECTED_SETTING(_profile, HistorySize);
-        OBSERVABLE_PROJECTED_SETTING(_profile, SnapOnInput);
-        OBSERVABLE_PROJECTED_SETTING(_profile, AltGrAliasing);
-        OBSERVABLE_PROJECTED_SETTING(_profile, BellStyle);
-        OBSERVABLE_PROJECTED_SETTING(_profile, UseAtlasEngine);
-
-    private:
-        Model::Profile _profile;
-        winrt::guid _originalProfileGuid;
-        winrt::hstring _lastBgImagePath;
-        winrt::hstring _lastStartingDirectoryPath;
-        Editor::AppearanceViewModel _defaultAppearanceViewModel;
-
-        static Windows::Foundation::Collections::IObservableVector<Editor::Font> _MonospaceFontList;
-        static Windows::Foundation::Collections::IObservableVector<Editor::Font> _FontList;
-
-        static Editor::Font _GetFont(com_ptr<IDWriteLocalizedStrings> localizedFamilyNames);
-
-        Model::CascadiaSettings _appSettings;
-        Editor::AppearanceViewModel _unfocusedAppearanceViewModel;
-    };
-
-    struct DeleteProfileEventArgs :
-        public DeleteProfileEventArgsT<DeleteProfileEventArgs>
-    {
-    public:
-        DeleteProfileEventArgs(guid profileGuid) :
-            _ProfileGuid(profileGuid) {}
-
-        guid ProfileGuid() const noexcept { return _ProfileGuid; }
-
-    private:
-        guid _ProfileGuid{};
-    };
-
     struct ProfilePageNavigationState : ProfilePageNavigationStateT<ProfilePageNavigationState>
     {
     public:
         ProfilePageNavigationState(const Editor::ProfileViewModel& viewModel,
                                    const Windows::Foundation::Collections::IMapView<hstring, Model::ColorScheme>& schemes,
-                                   const Editor::ProfilePageNavigationState& lastState,
                                    const IHostedInWindow& windowRoot) :
-            _Profile{ viewModel },
-            _Schemes{ schemes },
-            _WindowRoot{ windowRoot }
+            _Profile{ viewModel }
         {
-            // If there was a previous nav state copy the selected pivot from it.
-            if (lastState)
-            {
-                _LastActivePivot = lastState.LastActivePivot();
-            }
+            auto profile{ winrt::get_self<ProfileViewModel>(viewModel) };
```

Comment:
FWIW doing this _when a navigation state is constructed_ is unusual -- this should happen _to_ the VM after the navigation has completed.

Nav state should only be used to communicate incoming data to the view/viewmodel, not to actually enact change on it.

Reply:
That's fair. Adding this as a comment under the main MVVM issue.